### PR TITLE
Improve macos tray removal code

### PR DESCRIPTION
**Pull request by cfergeau**
This factors duplicated code by reusing existing launchd helpers, and rewords a bit the messages.

### CODE REVIEW

**Review by anjannath — APPROVED** [review]

File: pkg/crc/preflight/preflight_darwin.go [hunk 1]
```diff
@@ -2,14 +2,12 @@ package preflight
...
 	"fmt"
-	"os"
-	"path/filepath"
 
 	"github.com/code-ready/crc/pkg/crc/constants"
 	"github.com/code-ready/crc/pkg/crc/network"
 	"github.com/code-ready/crc/pkg/crc/preset"
 	crcpreset "github.com/code-ready/crc/pkg/crc/preset"
-	crcos "github.com/code-ready/crc/pkg/os"
+	"github.com/code-ready/crc/pkg/os/launchd"
```
Comment (gbraad) [inline pkg/crc/preflight/preflight_darwin.go:10]: this already exists or is missing from the PR?

Comment (cfergeau) [reply pkg/crc/preflight/preflight_darwin.go:10]: This already exists https://github.com/code-ready/crc/blob/main/pkg/os/launchd/launchd_darwin.go
(should have been removed a while ago, but was forgotten, and it's being reused in https://github.com/code-ready/crc/pull/3072 so we'll keep it)

**Comment by cfergeau** [comment]
CI seems to be failing because the cluster is too slow to start:
```
     Given checking that CRC is running           # crcsuite.go:271 -> github.com/code-ready/crc/test/e2e/crcsuite.CheckCRCStatus
    output did not match. Expected: '.*OpenShift: .*Running \(v\d+\.\d+\.\d+.*\).*', Actual: 'CRC VM:          Running
OpenShift:       Starting (v4.10.3) 
```

**Pull request merged**